lib/validation/validator.js

```diff
@@ -1,28 +1,48 @@
 import {fileURLToPath} from "node:url";
 import {readFile} from "node:fs/promises";
 
-async function loadSchema(schemaPath) {
-	const filePath = schemaPath.replace("http://ui5.sap/schema/", "");
-	const schemaFile = await readFile(
-		fileURLToPath(new URL(`./schema/${filePath}`, import.meta.url)), {encoding: "utf8"}
-	);
-	return JSON.parse(schemaFile);
-}
+/**
+ * @module @ui5/project/validation/validator
+ * @description A collection of validation related APIs
+ * @public
+ */
+
+/**
+ * @enum {string}
+ * @public
+ * @readonly
+ */
+export const SCHEMA_VARIANTS = {
+	"ui5": "ui5.json",
+	"ui5-workspace": "ui5-workspace.json"
+};
 
 class Validator {
-	constructor({Ajv, ajvErrors}) {
+	constructor({Ajv, ajvErrors, schemaName}) {
+		if (!schemaName && !SCHEMA_VARIANTS[schemaName]) {
+			throw new Error(
+				`"schemaName" is missing or incorrect. The available schemaName variants are ${Object.keys(
+					SCHEMA_VARIANTS
+				)}`
```

Comment:
```suggestion
				)}.join(", ")`
```

For better readability?